rc/tools/spell.kak

```diff
@@ -87,57 +87,60 @@ define-command spell-clear %{
     unset-option buffer spell_regions
 }
 
-define-command spell-next %{ evaluate-commands %sh{
+define-command spell-rel-jump -hidden -params 0..1 %{ evaluate-commands %sh{
     anchor_line="${kak_selection_desc%%.*}"
     anchor_col="${kak_selection_desc%%,*}"
     anchor_col="${anchor_col##*.}"
 
-    start_first="${kak_opt_spell_regions%%|*}"
-    start_first="${start_first#* }"
+    spell_first="${kak_opt_spell_regions%%|*}"
+    spell_first="${spell_first#* }"
+    spell_last="${kak_opt_spell_regions##* }"
+    spell_last="${spell_last%|*}"
 
     # Make sure properly formatted selection descriptions are in `%opt{spell_regions}`
-    if ! printf %s "${start_first}" | grep -qE '^[0-9]+\.[0-9]+,[0-9]+\.[0-9]+$'; then
+    if ! printf %s "${spell_first}" | grep -qE '^[0-9]+\.[0-9]+,[0-9]+\.[0-9]+$'; then
         exit
     fi
 
-    printf %s "${kak_opt_spell_regions#* }" | awk -v start_first="${start_first}" \
-                                                  -v anchor_line="${anchor_line}" \
-                                                  -v anchor_col="${anchor_col}" '
-        BEGIN {
-            anchor_line = int(anchor_line)
-            anchor_col = int(anchor_col)
-        }
-
-        {
-            for (i = 1; i <= NF; i++) {
-                sel = $i
-                sub(/\|.+$/, "", sel)
+    if [ "$1" = "-rev" ]; then
+        rev=1
+    elif [ -n "$1" ]; then
+        echo "fail -- Unrecognised parameter $(kakquote "$1")"
+    fi
 
-                start_line = sel
-                sub(/\..+$/, "", start_line)
-                start_line = int(start_line)
+    printf %s "${kak_opt_spell_regions#* }" | \
+        awk -v spell_first="${spell_first}" -v spell_last="${spell_last}" \
+            -v anchor_line="${anchor_line}" -v anchor_col="${anchor_col}" \
```

Comment:
(unrelated:) I don't really understand why we use the anchor's coordinates and not the cursor's,
or the minimum/maximum of cursor/anchor.
The current behavior is that given

	  mispelled    tpo
	A------------C

spell-next will jump to "mispelled", which is weird.

Reply:
good point--I find that weird as well. definitely wouldn't be hard to implement

Reply:
hmm. I think the best solution to this would be to, for spell-next, actually check the end of the word with the end of the current selection, while for spell-prev, check the start of the word with the start of the current selection, opposite of intuition.

This would cause the following situations, which to me, feels the most intuitive when searching for the next misspelled word:

When using `:spell-next`:
```
word misspelled1 word misspelled2 word
   X-------X
     X---------X
```
```
word misspelled1 word misspelled2 word
         X---------X
                      X---------X
```

When using `:spell-prev`:
```
word misspelled1 word misspelled2 word
         X---------X
     X---------X
```
```
word misspelled1 word misspelled2 word
                  X--------X
     X---------X
```